### PR TITLE
Fake task analysis using current fake pana and dartdoc.

### PR DESCRIPTION
**Pull request by isoos**
- This is a requirement for #6834, submitting it separately.
- The largest change seems to be the retracted version template.

### CODE REVIEW

File: app/lib/fake/backend/fake_pub_worker.dart [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+              Future<void> addFileAsStringGzipped(
+                  String path, String content) async {
+                final stream =
+                    Stream.fromIterable([gzip.encode(utf8.encode(content))]);
+                await builder.addFile(path, stream);
+              }
+
+              for (final e in dartdocFiles.entries) {
+                await addFileAsStringGzipped(e.key, e.value);
```
Comment (jonasfj) [inline app/lib/fake/backend/fake_pub_worker.dart:128]: The `.html` files created by `pub_worker` are actually JSON files :rofl: 

This happened when we moved the postprocessing/sanitizing into the sandbox

File: app/lib/fake/backend/fake_dartdoc_runner.dart [hunk 2]
```diff
@@ -52,26 +52,29 @@ class FakeDartdocRunner implements DartdocRunner {
...
+    }
 
     return DartdocRunnerResult(
       args: ['fake_dartdoc', '--input', pkgPath, '--output', outputDir],
       processResult: PanaProcessResult(0, 0, 'OK', ''),
     );
   }
 }
+
+Map<String, String> fakeDartdocFiles(String package, version) {
```
Comment (jonasfj) [inline app/lib/fake/backend/fake_dartdoc_runner.dart:67]: wouldn't it be easier to copy / paste this, than have cross dependencies...

File: app/lib/fake/backend/fake_pub_worker.dart (hunk 1 above)
Comment (jonasfj) [reply app/lib/fake/backend/fake_pub_worker.dart:128]: We can probably just hardcode some JSON that fits the DartDocPage pattern

Comment (isoos) [reply app/lib/fake/backend/fake_pub_worker.dart:130]: Updated. PTAL.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**